benchmarks/test_tensordot.py

```diff
@@ -14,7 +14,11 @@ def get_sides_ids(param):
     return f"{m=}-{n=}-{p=}-{q=}"
 
 
-@pytest.fixture(params=itertools.product([10, 20, 50], [10, 20, 50], [10, 20, 50], [10, 20, 50]), ids=get_sides_ids)
+@pytest.fixture(
+    params=itertools.product([10, 20, 50], [10, 20, 50], [10, 20, 50], [10, 20, 50]),
```

Comment:
Uhh, we're doing `3 ** 4 = 81` permutations here. We might need to put this into a list, shuffle it, and pick the first 10 maybe, but the shuffle needs to be deterministic otherwise we have benchmarks disappearing and appearing.